bundles/org.openhab.binding.vesync/src/main/java/org/openhab/binding/vesync/internal/handlers/VeSyncDeviceOutletHandler.java

```diff
@@ -0,0 +1,238 @@
+                String[] responseStrings = responses.split("/?");
+                responseStatus = responseStrings[0];
+                responseEnergyHistory = responseStrings[1];
+            }
+
+            if (responseStatus.equals(EMPTY_STRING) || responseEnergyHistory.equals(EMPTY_STRING)) {
+                return;
+            }
+
+            outletStatus = VeSyncConstants.GSON.fromJson(responseStatus, VeSyncV2BypassOutletStatus.class);
+            energyHistory = VeSyncConstants.GSON.fromJson(responseEnergyHistory, VeSyncV2BypassEnergyHistory.class);
+
+            if (outletStatus == null || energyHistory == null) {
+                return;
+            }
+            if (!cachedDataUsed) {
+                cachedResponse.putValue(responseStatus + "?" + responseEnergyHistory);
+            }
+        }
+
+        // Bail and update the status of the thing - it will be updated to online by the next search
+        // that detects it is online.
+        if (outletStatus.isMsgDeviceOffline()) {
+            updateStatus(ThingStatus.OFFLINE);
+            return;
+        } else if (outletStatus.isMsgSuccess()) {
+            updateStatus(ThingStatus.ONLINE);
+        }
+
+        if (!"0".equals(outletStatus.getCode())) {
+            logger.warn("{}", getLocalizedText("warning.device.unexpected-resp-for-wifi-outlet"));
+            return;
+        }
+
+        updateState(DEVICE_CHANNEL_ENABLED,
+                OnOffType.from(MODE_ON.equals(outletStatus.outletResult.result.getDeviceStatus())));
+        updateState(DEVICE_CHANNEL_CURRENT, new QuantityType<>(outletStatus.outletResult.result.current, Units.AMPERE));
+        updateState(DEVICE_CHANNEL_VOLTAGE, new QuantityType<>(outletStatus.outletResult.result.voltage, Units.VOLT));
+        updateState(DEVICE_CHANNEL_ENERGY, new QuantityType<>(outletStatus.outletResult.result.energy, Units.WATT));
+        updateState(DEVICE_CHANNEL_POWER,
+                new QuantityType<>(outletStatus.outletResult.result.power, MetricPrefix.KILO(Units.WATT_HOUR)));
+        updateState(DEVICE_CHANNEL_HIGHEST_VOLTAGE,
+                new QuantityType<>(outletStatus.outletResult.result.highestVoltage, Units.VOLT));
+        updateState(DEVICE_CHANNEL_VOLTAGE_PT_STATUS, OnOffType.from(outletStatus.outletResult.result.voltagePTStatus));
+        updateState(DEVICE_CHANNEL_ENERGY_WEEK, new DecimalType(getEnergy(energyHistory, 7)));
+        updateState(DEVICE_CHANNEL_ENERGY_MONTH, new DecimalType(getEnergy(energyHistory, 30)));
+    }
+
+    private static long getTimestampForToday() throws ParseException {
+        SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+        Date todayDate = new Date();
+        String todayString = simpleDateFormat.format(todayDate);
+        String endString = todayString.replaceFirst("/d/d:/d/d:/dd", "02:00:00");
+        return simpleDateFormat.parse(endString).getTime() / 1000;
+    }
```

Comment:
Why is this hardcoded to 02:00:00 ?

This method seems to return the amount of seconds between today and epoch. And the timezone is ambigous. In that case i would be better to use `Instant`. If you need to add a timezoen specific to the device configuration, it is better to provide that configuration as a parameter to the method. 


```suggestion
    private static long getTimestampForToday() throws ParseException {
        Instant instant = Instant.now().truncatedTo(ChronoUnit.DAYS);
        return instant.toEpochMilli() / 1000;
    }
```

Reply:
Fixed in 7893d2e